tests/Unit/Domain/Test_BlockNeighbor.cpp

```diff
@@ -0,0 +1,40 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "Domain/BlockNeighbor.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+TEST_CASE("Unit.Domain.BlockNeighbor", "[Domain][Unit]") {
+  // Test default constructor:
+  BlockNeighbor<3> test_block_neighbor;
```

Comment:
We should have actual `CHECK(...)` calls after the construction to make sure the default constructor does what we expect it to. If it's only used for Charm++ serialization then I don't think the `CHECK`s are needed, but a comment by the test would be a helpful reminder. What I have in mind is something like `Test default constructor, only used for Charm++ serialization so no CHECK calls`